src/charm.py

```diff
@@ -85,6 +86,12 @@ def _configure(self, event: EventBase) -> None:
                 f"The following configurations are not valid: {invalid_configs}"
             )
             return
+        try:
+            self._kubernetes_multus.is_ready()
+        except KubernetesMultusError as err:
+            self.unit.status = BlockedStatus(err.message)
```

Comment:
Should we be setting the status to Blocked? Looking at the implementation of `is_ready`, when it returns `False`, it may just as well mean that we're still waiting for the Pod or the StatefulSet to be patched. I think how we handle this check in the UPF (`WaitingStatus`) is a right thing to do.
For checking Multus availability we could potentially look for it under `/opt/cni/bin` (we should probably check if there are any more elegant alternatives tho). If not, then Blocked status is fine.

Reply:
Actually UPF suffers from this same issue. If Multus is not enabled, `is_ready` throws an exception (since k8s client returns 404 Not Found for the NAD resource).
To check availability without invoking `is_ready`, we could assert the existence of the NAD resource directly in the charm.
Regarding Blocked or Waiting, I don't have a strong opinion on this.

Reply:
I like the idea of checking NAD CRD. It that case, I'd vote for making it a separate check which we would run `on_install`, similarly to CPU flags check in the UPF. WDYT?

Reply:
Setting the status in the `on_install` hook would not work, as `install` event will always trigger `config-changed` event which will fail for the same reason. We agreed to check it in the `config_changed` hook and also start observing `update-status` event.

Reply:
@gruyaume what do you think? I tried moving the check in the `install` hook raising an exception. The output of `juju status` will be the following:
```
Unit                  Workload  Agent  Address      Ports  Message
sdcore-router-k8s/0*  error     idle   10.1.142.93         hook failed: "install"
```
The user should use `juju debug-log` to see why it failed, enable multus and then run `juju resolved sdcore-router-k8s/0` to have the charm proceed.

Reply:
Yeah, message kinda sucks, but if we provide a detailed description of the problem in the logs (Multus is disabled, enable Multus and run juju resolved), then I think it's safer than relying on `update-status`.

Reply:
I'm ok with using update status. This is quite useful as we know it comes back every X minutes and we have a scheduled moment where we can compare the state of the world with our config and make the necessary changes if needed.